scripts/install-build-tools.sh

```diff
@@ -14,30 +15,218 @@ if (( $EUID != 0 )); then
+  for PY_VERS in "${PYTHON_VERSIONS[@]}"; do
+    PYTHON_PATH="${BREW_ROOT}/bin/python${PY_VERS}"
+    # save the installed version if found
+    if [ -e "${PYTHON_PATH}" ]; then
+      PY_INSTALLED=${PY_VERS}
+      break
+    fi
+  done
+
+  # install valid python version if not installed yet
+  if [ -z "$PY_INSTALLED" ]; then
+    for PY_VERS in "${PYTHON_VERSIONS[@]}"; do
+      FULL_PY="python${PY_VERS}"
+      PYTHON_PATH="${BREW_ROOT}/bin/${FULL_PY}"
+      # try to install python version from homebrew and verify installation
+      if brew install "${FULL_PY}"; then
+        echo "${FULL_PY} installed successfully"
+        PY_INSTALLED=${PY_VERS}
+        break
+      else
+        # don't exit as we can try on another version if not the last one
+        echo "${FULL_PY} installation failed"
+      fi
+    done
+  fi
+
+  # leave if no valid python version is installed
+  if [ -z "$PY_INSTALLED" ]; then
+    echo "Python3 install with brew failed, attempted on these versions: ${PYTHON_VERSIONS[@]}"
+    exit 1
+  fi
+
+  # create virtual environment and install python packages for the valid python version
+  create_venv_install_python "${PYTHON_PATH}" ${PY_INSTALLED}
+
+  echo "To activate the virtual env to run python, run 'source .py_venv/bin/activate'"
+
+# Linux install with apt
+elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  # avoids getting stuck on interactive prompts which is essential for CI/CD
+  export DEBIAN_FRONTEND=noninteractive
+  $SUDO apt update -y
+  $SUDO apt upgrade -y
+  $SUDO apt install -y build-essential wget cmake libgtest-dev libbenchmark-dev \
+    lcov git software-properties-common rsync unzip bc
 
-  wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | $SUDO apt-key add -
-  $SUDO add-apt-repository "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main"
+  # Add LLVM GPG key (apt-key is deprecated in Ubuntu 21.04+ so using gpg)
+  wget -qO - https://apt.llvm.org/llvm-snapshot.gpg.key | \
+    gpg --dearmor -o /usr/share/keyrings/llvm-archive-keyring.gpg
+  echo "deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main" | \
+    $SUDO tee /etc/apt/sources.list.d/llvm.list
+
+  $SUDO apt update -y
   $SUDO apt install -y clang-format-14 clang-tidy-14
-  $SUDO ln -s -f $(which clang-format-14) /usr/local/bin/clang-format
-  $SUDO ln -s -f $(which clang-tidy-14) /usr/local/bin/clang-tidy
+  $SUDO ln -sf $(which clang-format-14) /usr/local/bin/clang-format
+  $SUDO ln -sf $(which clang-tidy-14) /usr/local/bin/clang-tidy
+
+  # check if supported version of python3 is already installed, and save the version
+  PYTHON_PATH=''
+  for PY_VERS in "${PYTHON_VERSIONS[@]}"; do
+    PYTHON_PATH="/usr/bin/python${PY_VERS}"
+    # save the installed version if found
+    if [ -e "${PYTHON_PATH}" ]; then
+      PY_INSTALLED=${PY_VERS}
+      break
+    fi
+  done
+
+  # install valid python version if not installed yet
+  if [-z  "$PY_INSTALLED" ]; then
```

Comment:
`[` is a command, not special syntax, so there must be a space separating `[` from any of the expressions it is evaluating.

Reply:
ACK, just fixed thanks